engine/access/state_stream/backend/handler.go

```diff
@@ -215,3 +217,260 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+func (h *Handler) createAccountStatusesFilter(statusFilter *executiondata.StatusFilter) (state_stream.EventFilter, error) {
+	filter := state_stream.EventFilter{}
+	if statusFilter != nil {
+		var err error
+		filter, err = state_stream.NewEventFilter(
+			h.eventFilterConfig,
+			h.chain,
+			statusFilter.GetEventType(),
+			[]string{},
+			[]string{},
+		)
+		if err != nil {
+			return filter, status.Errorf(codes.InvalidArgument, "invalid event filter: %v", err)
+		}
+
+		addresses := statusFilter.GetAddress()
+		if len(addresses) > 0 {
+			var fieldFilters []state_stream.FieldFilter
+			for _, address := range addresses {
+				fieldFilters = append(fieldFilters, state_stream.FieldFilter{
+					FieldName:   "address",
+					TargetValue: address,
+				})
+			}
+
+			for eventType := range filter.EventTypes {
+				filter.EventFieldFilters[eventType] = fieldFilters
+			}
+		}
+	}
+
+	return filter, nil
+}
+
+func convertAccountsStatusesResults(eventVersion entities.EventEncodingVersion, resp *AccountStatusesResponse) ([]*executiondata.SubscribeAccountStatusesResponse_Result, error) {
+	var results []*executiondata.SubscribeAccountStatusesResponse_Result
+	for address, events := range resp.AccountEvents {
+		convertedEvent, err := convert.EventsToMessagesWithEncodingConversion(events, entities.EventEncodingVersion_CCF_V0, eventVersion)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "could not convert events to entity: %v", err)
+		}
+
+		results = append(results, &executiondata.SubscribeAccountStatusesResponse_Result{
+			Address: []byte(address),
```

Comment:
what is `address` here? We want to return the binary representation of the hex data. We probably need to use `flow.HexToAddress(address).Bytes()` or something like that.